packages/email-petition/src/email-petition.tsx

```diff
@@ -200,12 +226,43 @@ export function EmailPetition(props: {
 
   const queryClient = useMemo(() => new QueryClient(), []);
 
-  // Clear zip code if not in US to avoid validation errors when this field must be blank anyway.
+  // Call API when zip code changes
   useEffect(() => {
-    if (outsideUS) {
-      setValue("zip", "");
-      clearErrors(["zip", "city"]);
+    if (
+      zip &&
+      !outsideUS &&
+      zip.length === 5 &&
+      props.citySelectionMode === "autocompleteTextbox"
+    ) {
+      fetchCityByZip(zip);
+    } else if (
+      zip &&
+      !outsideUS &&
+      zip.length === 5 &&
+      props.citySelectionMode === "sonomaCountyDropdown"
+    ) {
+      // For non-Google Maps apps, only show city field if ZIP is in SonomaCities
+      if (zip in SonomaCities) {
+        setHideCity(false);
+        setUserInteractedWithCityField(false);
+      } else {
+        setValue("city", "");
+        setHideCity(true);
+        setUserInteractedWithCityField(false);
+      }
+    } else {
+      setValue("city", "");
+      setHideCity(true);
+      setUserInteractedWithCityField(false); // Reset user interaction when ZIP changes
     }
+  }, [zip, outsideUS, setValue]);
+
+  useEffect(() => {
+    setValue("zip", "");
+    setValue("city", "");
+    setCoordinates(null);
+    clearErrors(["zip", "city"]);
+    setUserInteractedWithCityField(false); // Reset user interaction when outsideUS changes
```

Comment:
remove comment. feel free to add a comment for the whole "useEffect" if it would be helpful.